dexter/src/main/java/com/karumi/dexter/DexterInstance.java

```diff
@@ -114,6 +114,7 @@ void onCancelPermissionRequest(String permission) {
    */
   void requestPermission(String permission) {
     int permissionCode = getPermissionCodeForPermission(permission);
+    this.firstTimeRequested = !ActivityCompat.shouldShowRequestPermissionRationale(activity, permission);
```

Comment:
This won't work as expected, when the user *perma-denies* a permission it will report that it has been the first time requesting it:

<img width="1244" alt="screen shot 2015-11-18 at 10 49 16" src="https://cloud.githubusercontent.com/assets/3116415/11237764/0ec407c4-8de2-11e5-89c4-8115239a10a9.png">

The only safe way of doing this that I can think of is to store previous requests. We will probably need to store information in the shared preferences anyways if we want to support reporting *perma-denied* permissions as well.